src/main/java/seedu/address/model/food/Food.java

```diff
@@ -33,17 +42,52 @@ public Food(String name, Protein protein, Carbohydrate carbs, Fat fat) {
     public Food(String name, int proteinAmount, int carbsAmount, int fatAmount) {
         this(name, new Protein(proteinAmount), new Carbohydrate(carbsAmount), new Fat(fatAmount));
     }
+
+    private boolean isValidName(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
     // getters : make when needed
+    private String getName() {
+        return this.name;
+    }
 
-    // isSameFood : make when needed
+    private Protein getProtein() {
+        return this.protein;
+    }
 
-    // Displays
-    // name + PCF details + total calories
+    private Carbohydrate getCarbs() {
+        return this.carbs;
+    }
+
+    private Fat getFat() {
+        return this.fat;
+    }
 
+    /**
+     * Compare if two food item are the same
+     * Compare criteria: same name (case sensitive) and same nutrition values
+     * @param otherFood Other food to be compared with this food
+     * @return True if the 2 are the same
+     */
+    public boolean isSameFood(Food otherFood) {
```

Comment:
I think we should `@override` the `object.equals` method instead. Having 2 methods claiming to do the same thing will invite bugs.

Reply:
Ok I will change it